src/main/java/com/linkedin/kmf/services/ProduceService.java

```diff
@@ -191,12 +191,12 @@ public boolean isRunning() {
     return _running.get();
   }
 
-  private class ProduceMetrics {
+  protected class ProduceMetrics {
```

Comment:
Probably the visibility of this class does not need to change if the other refactorings are implemented.